src/Aspire.Dashboard/Extensions/FluentUiExtensions.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Dashboard.Extensions;
+
+internal static class FluentUiExtensions
+{
+    public static Dictionary<string, object> GetClipboardCopyAdditionalAttributes(string buttonId, string? text, string? precopy, string? postcopy, params (string Attribute, object Value)[] additionalAttributes)
+    {
+        var attributes = new Dictionary<string, object>
+        {
+            { "data-text", text ?? string.Empty },
+            { "data-precopy", precopy ?? string.Empty },
+            { "data-postcopy", postcopy ?? string.Empty },
+            { "onclick", $"buttonCopyTextToClipboard('{buttonId}')" }
```

Comment:
Instead of passing an id, I think you can pass the element directly as `this`.
```suggestion
            { "onclick", "buttonCopyTextToClipboard(this)" }
```
Then update buttonCopyTextToClipboard to get attributes from the passed in button element.